core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/SonarLintInputFile.java

```diff
@@ -80,7 +84,7 @@ private static UnsupportedOperationException unsupported() {
 
   @Override
   public Charset charset() {
-    Charset charset = clientInputFile.charset();
+    Charset charset = clientInputFile.getCharset();
     return charset != null ? charset : Charset.defaultCharset();
```

Comment:
For now clientInputFile.getCharset() is @CheckForNull. We could think about changing that later.